tiledb/sm/serialization/array_schema.cc

```diff
@@ -242,9 +258,12 @@ Status attribute_from_capnp(
   // Set filter pipelines.
   if (attribute_reader.hasFilterPipeline()) {
     auto filter_pipeline_reader = attribute_reader.getFilterPipeline();
-    tdb_unique_ptr<FilterPipeline> filters;
-    RETURN_NOT_OK(filter_pipeline_from_capnp(filter_pipeline_reader, &filters));
-    RETURN_NOT_OK((*attribute)->set_filter_pipeline(filters.get()));
+    auto&& [st_fp, filters]{filter_pipeline_from_capnp(filter_pipeline_reader)};
+    if (!st_fp.ok()) {
```

Comment:
Use `RETURN_NOT_OKAY` macro for getting this and other statuses.

Reply:
wait, I thought you weren't supposed to do that?

Reply:
`RETURN_NOT_OK` is preferred and safe. Please swap all okay checks in this PR for the macro.

Reply:
RETURN_NOT_OK only returns a status, when we want to return a tuple of status and optional shared pointer. The code won't compile if I switch to using the macro.